experimental/dds/tree2/src/shared-tree-core/revisionTagCodecs.ts

```diff
@@ -3,14 +3,30 @@
  * Licensed under the MIT License.
  */
 
+import { assert } from "@fluidframework/core-utils";
+import { IIdCompressor, SessionId } from "@fluidframework/runtime-definitions";
 import { IJsonCodec } from "../codec";
 import { EncodedRevisionTag, RevisionTag } from "../core";
 
 export class RevisionTagCodec implements IJsonCodec<RevisionTag, EncodedRevisionTag> {
+	public constructor(private readonly idCompressor?: IIdCompressor) {}
+
 	public encode(tag: RevisionTag) {
-		return tag as unknown as EncodedRevisionTag;
+		assert(
+			this.idCompressor !== undefined,
+			"IdCompressor must be provided to encode a revision tag",
+		);
+		return this.idCompressor.normalizeToOpSpace(tag) as EncodedRevisionTag;
 	}
-	public decode(tag: EncodedRevisionTag) {
-		return tag as unknown as RevisionTag;
+	public decode(tag: EncodedRevisionTag, originatorId?: SessionId) {
+		assert(
```

Comment:
Ideally this wouldn't be guarded by an assertion and instead `originatorId` would be required in any place that needs it. Didn't have time to clean up with a better solution